pyronear_ds/datasets/utils.py

```diff
@@ -16,3 +31,177 @@ def get_intersection_range(ts1: pd.Series, ts2: pd.Series) -> pd.DatetimeIndex:
         raise ValueError("Extracts do not have intersecting date range")
 
     return pd.date_range(time_range1, time_range2)
+
+
+def url_retrieve(url: str, timeout: Optional[float] = None) -> bytes:
+    """Retrives and pass the content of an URL request
+
+    Args:
+        url: URL to request
+        timeout: number of seconds before the request times out. Defaults to 4.
+
+    Raises:
+        requests.exceptions.ConnectionError:
+
+    Return:
+        Content of the response
+    """
+
+    response = requests.get(url, timeout=timeout, allow_redirects=True)
+    if response.status_code != 200:
+        raise requests.exceptions.ConnectionError(f'Error code {response.status_code} - could not download {url}')
+    return response.content
+
+
+def get_fname(url: str) -> Tuple[str, str, str]:
+    """Find file name, extension and compression of an archive located by an URL
+
+    Args:
+        url: URL of the compressed archive
+
+    Raises:
+        ValueError: if URL contains more than one extension
+        ValueError: if URL contains more than one compression format
+
+    Returns:
+        A tuple containing the base file name, extension and compression format
+    """
+
+    supported_compressions = ["tar", "gz", "zip"]
+    supported_extensions = ["csv", "geojson", "shp", "shx", "nc"]
+
+    archive_name = urlparse(url).path.rpartition('/')[-1]
+
+    base = archive_name.split('.')[0]
+
+    list_extensions = list(set(supported_extensions) & set(archive_name.split('.')))
+    list_compressions = list(set(supported_compressions) & set(archive_name.split('.')))
+
+    if len(list_extensions) == 0:
+        extension = None
+    elif len(list_extensions) == 1:
+        extension = list_extensions[0]
+    else:
+        raise ValueError(f'Error {url} contains more than one extension')
+
+    if len(list_compressions) == 0:
+        compression = None
+
+    elif len(list_compressions) == 1:
+        compression = list_compressions[0]
+
+    elif len(list_compressions) == 2:
+        compression = "tar.gz"
+
+    else:
+        raise ValueError(f'Error {url} contains more than one compression format')
+
+    return (base, extension, compression)
+
+
+def download(url: str, default_extension: str, unzip: Optional[bool] = True, destination: Optional[str] = './tmp'):
+    """Helper function for downloading, unzipping and saving compressed file from a given URL.
+
+    Args:
+        url: URL of the compressed archive
+        default_extension: extension of the archive
+        unzip: whether archive should be unzipped. Defaults to True.
+        destination: folder where the file should be saved. Defaults to '.'.
+    """
+# TODO Write case tests for zip, tar.gz, gz and uncompressed files
+# Check if the destination directory is created each if not exist
+# Check if the file are  download
+# Add print and logging statement add
+
+    base, extension, compression = get_fname(url)
+
+    if unzip is True and compression == "zip":
```

Comment:
Since `unzip` is already a bool, this can be changed to :
`if unzip and compression == "zip":`

Reply:
Thanks